File: trunk/esper/src/main/java/net/esper/eql/join/QueryStrategy.java
package net.esper.eql.join;

import net.esper.event.EventBean;
import net.esper.collection.MultiKey;

import java.util.Set;

/**
 * Encapsulates the strategy use to resolve the events for a stream into a tuples of events in a join.
 */
public interface QueryStrategy
{
    /**
     * Look up events returning tuples of joined events.
     * @param lookupEvents - events to use to perform the join
     * @param joinSet - result join tuples of events 
     */
    public void lookup(EventBean[] lookupEvents, Set<MultiKey<EventBean>> joinSet);
}

File: NEsper/eql/join/QueryStrategy.cs
using System;
using System.Collections.Generic;

using net.esper.events;
using net.esper.collection;
using net.esper.compat;

namespace net.esper.eql.join
{
	/// <summary>
    /// Encapsulates the strategy use to resolve the events for a stream into a tuples of events in a join.
    /// </summary>

    public interface QueryStrategy
    {
        /// <summary> Look up events returning tuples of joined events.</summary>
        /// <param name="lookupEvents">events to use to perform the join
        /// </param>
        /// <param name="joinSet">result join tuples of events 
        /// </param>

        void Lookup(EventBean[] lookupEvents, Set<MultiKey<EventBean>> joinSet);
    }
}
